### DISCUSSION TITLE
Scrolling title in HUD

### DISCUSSION THREAD

**Discussion by Vasmares** [BimmerGestalt/AAIdrive#668] [Ideas]
Hi,

I just updated to the nightly build of 1.4.1 from 10th Feb 2023 and only now realized that the titles are now scrolling if they are too long to fit.

This is brilliant. Awsome Idea. Much Praise.

Is there a way to implement this for the HUD as well ?

When selecting a new track in the HUD i dont just skip forward to the track that I like, I can see the entire list of tracks that are in the current playlist. So I can explicitly select a title.

When the title is too long, you cant read it properly, because of the other information in the HUD (speed, signs, navigation etc.)
It would be awsome if those would scroll as well.

![20230211_013552](https://user-images.githubusercontent.com/10863304/218252860-369f2524-8df8-4ee6-91d3-76a86c51ff0f.jpg)

**Comment by hufman** [comment]
The autoscrolling text is implemented manually, by repeatedly sending different parts of the string to the car, instead of being a feature provided by the car. The main screen tells AAIdrive what screen it is on, so it only animates the text while it is visible.
The HUD playlist, however, does not provide any feedback to AAIdrive when the user is scrolling through it, so there is no signal to know when to start animating the text. I am also not sure about the car's behavior if that playlist is being replaced constantly with different text while the user is scrolling through it.

**Reply by Vasmares** [reply]
I get what you mean.
Also, the track list ist just there for about 3 seconds.
So it wouldnt make any sense to start scrolling.

One can still use the center screen to achive the same result :D

I was just being lazy.

Thank you for your reply.
